.github/workflows/ci-extended.yml

```diff
@@ -44,10 +44,20 @@ jobs:
           path: tst/regression/gold_standard/
           key: gold-standard
 
+      - name: Set vars based on matrix
+        id: cmake-vars
+        run: |
+          if ${{ matrix.device == 'host' }}; then
+            echo "enable_asan=ON" >> $GITHUB_OUTPUT
+          else
+            echo "enable_asan=OFF" >> $GITHUB_OUTPUT
+          fi
+
       - name: Configure
         run: |
           cmake -B build \
             -DCMAKE_BUILD_TYPE=Release \
+            -DENABLE_ASAN=${{ steps.cmake-vars.outputs.enable_asan }} \
```

Comment:
Is that a common pattern?

Might sth like
```bash
-DENABLE_ASAN=${${{ matrix.device == 'host' }}^^} \
```
work, too?

Or if that one line logic doesn't work using an env var:
```
env:
  ENABLE_ASAN: ${{ matrix.device == 'host' }}
```
with (the `^^` to uppercase the lowercase result in the env var):
```
-DENABLE_ASAN=${ENABLE_ASAN^^} \
```

Reply:
Bash outputs `1` for true and `0` for false, so I don't think that works, unless CMake translates `1` to `ON`?

Reply:
I think it should 
"True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number (including floating point numbers)." from https://cmake.org/cmake/help/latest/command/if.html

But I think GitHub actions only returns (lowercase) `true` or `false` for expressions so that's why we need to convert it to uppercase.